src/snowflake/snowpark/session.py

```diff
@@ -3181,6 +3181,11 @@ def write_pandas(
                 f"Unsupported table type. Expected table types: {SUPPORTED_TABLE_TYPES}"
             )
 
+        if hasattr(df, "columns") and len(df.columns) == 0:
+            raise ValueError(
```

Comment:
what's the previous error class raised?
changing raised error class technically is a BCR change

Reply:
Good point, this is previously a programming error from connector that the create table sql will fail:
```
snowflake.connector.errors.ProgrammingError: 001003 (42000): 01bc8160-0810-2189-0001-dd4b5930ee17: SQL compilation error:
syntax error line 1 at position 89 unexpected ')'.
```